src/main/java/me/ryanhamshire/GriefPrevention/Claim.java

```diff
@@ -443,214 +395,184 @@ public boolean hasExplicitPermission(Player player, ClaimPermission level)
-        for (int i = 0; i < this.managers.size(); i++)
-        {
-            String managerID = this.managers.get(i);
-            if (player.getUniqueId().toString().equals(managerID)) return null;
+        return checkPermission(player, ClaimPermission.Build, new CompatBuildBreakEvent(material, true));
+    }
 
-            else if (managerID.startsWith("[") && managerID.endsWith("]"))
-            {
-                managerID = managerID.substring(1, managerID.length() - 1);
-                if (managerID == null || managerID.isEmpty()) continue;
-                if (player.hasPermission(managerID)) return null;
-            }
-        }
+    /**
+     * @deprecated Check {@link ClaimPermission#Access} with {@link #checkPermission(Player, ClaimPermission, Event)}.
+     * @param player the Player
+     * @return the denial message, or null if the action is allowed
+     */
+    @Deprecated
+    public String allowAccess(Player player)
+    {
+        return checkPermission(player, ClaimPermission.Access, null);
+    }
 
-        //permission inheritance for subdivisions
-        if (this.parent != null)
-        {
-            if (player.getUniqueId().equals(this.parent.ownerID))
-                return null;
-            if (!inheritNothing)
-                return this.parent.allowGrantPermission(player);
-        }
+    /**
+     * @deprecated Check {@link ClaimPermission#Inventory} with {@link #checkPermission(Player, ClaimPermission, Event)}.
+     * @param player the Player
+     * @return the denial message, or null if the action is allowed
+     */
+    @Deprecated
+    public String allowContainers(Player player)
+    {
+        return checkPermission(player, ClaimPermission.Inventory, null);
+    }
 
-        //generic error message
-        String reason = GriefPrevention.instance.dataStore.getMessage(Messages.NoPermissionTrust, this.getOwnerName());
-        if (player.hasPermission("griefprevention.ignoreclaims"))
-            reason += "  " + GriefPrevention.instance.dataStore.getMessage(Messages.IgnoreClaimsAdvertisement);
-        return reason;
+    /**
+     * @deprecated Check {@link ClaimPermission#Manage} with {@link #checkPermission(Player, ClaimPermission, Event)}.
+     * @param player the Player
+     * @return the denial message, or null if the action is allowed
+     */
+    @Deprecated
+    public String allowGrantPermission(Player player)
+    {
+        return checkPermission(player, ClaimPermission.Manage, null);
     }
 
     public ClaimPermission getPermission(String playerID)
     {
-        if (playerID == null || playerID.isEmpty())
-        {
-            return null;
-        }
+        if (playerID == null || playerID.isEmpty()) return null;
 
         return this.playerIDToClaimPermissionMap.get(playerID.toLowerCase());
     }
 
     //grants a permission for a player or the public
     public void setPermission(String playerID, ClaimPermission permissionLevel)
```

Comment:
Wasn't sure how we wanted to handle addons trying to set the new ClaimPermissions. This may actually be a beneficial change - privatizing Claim#managers is probably a good idea.